Explorer/Assets/Scripts/ECS/Unity/AvatarShape/Systems/AvatarShapeHandlerSystem.cs

```diff
@@ -0,0 +1,96 @@
+using DCL.Utilities;
+using ECS.LifeCycle;
+using ECS.LifeCycle.Components;
+using ECS.Prioritization.Components;
+using ECS.Unity.AvatarShape.Components;
+using ECS.Unity.Groups;
+using ECS.Unity.Transforms.Components;
+
+namespace ECS.Unity.AvatarShape.Systems
+{
+    [UpdateInGroup(typeof(ComponentInstantiationGroup))]
+    [LogCategory(ReportCategory.AVATAR)]
+    [ThrottlingEnabled]
+    public partial class AvatarShapeHandlerSystem : BaseUnityLoopSystem, IFinalizeWorldSystem
+    {
+        private WorldProxy globalWorld;
+        private ReleaseOnEntityDestroy releaseOnEntityDestroy;
+
+        public AvatarShapeHandlerSystem(World world, WorldProxy globalWorld) : base(world)
+        {
+            this.globalWorld = globalWorld;
+            releaseOnEntityDestroy = new ReleaseOnEntityDestroy(globalWorld);
+        }
+
+        protected override void Update(float t)
+        {
+            LoadAvatarShapeQuery(World);
+            UpdateAvatarShapeQuery(World);
+
+            HandleComponentRemovalQuery(World);
+            HandleEntityDestructionQuery(World);
+        }
+
+        [Query]
+        [None(typeof(SDKAvatarShapeComponent))]
+        private void LoadAvatarShape(in Entity entity, ref PBAvatarShape pbAvatarShape, ref PartitionComponent partitionComponent, ref TransformComponent transformComponent)
+        {
+            Entity globalWorldEntity = globalWorld.Create(pbAvatarShape, partitionComponent, transformComponent);
+            World.Add(entity, new SDKAvatarShapeComponent(globalWorldEntity));
+        }
+
+        [Query]
+        private void UpdateAvatarShape(ref PBAvatarShape pbAvatarShape, ref SDKAvatarShapeComponent sdkAvatarShapeComponent)
+        {
+            if (!pbAvatarShape.IsDirty)
+                return;
+
+            globalWorld.Add(sdkAvatarShapeComponent.globalWorldEntity, pbAvatarShape);
+        }
+
+        [Query]
+        [None(typeof(PBAvatarShape), typeof(DeleteEntityIntention))]
+        private void HandleComponentRemoval(in Entity entity, ref SDKAvatarShapeComponent sdkAvatarShapeComponent)
+        {
+            // If the component is removed at scene-world, the global-world representation should disappear entirely
+            globalWorld.Add(sdkAvatarShapeComponent.globalWorldEntity, new DeleteEntityIntention());
+
+            World.Remove<SDKAvatarShapeComponent>(entity);
+        }
+
+        [Query]
+        [All(typeof(DeleteEntityIntention))]
+        private void HandleEntityDestruction(in Entity entity, ref SDKAvatarShapeComponent sdkAvatarShapeComponent)
+        {
+            World.Remove<SDKAvatarShapeComponent>(entity);
+            globalWorld.Add(sdkAvatarShapeComponent.globalWorldEntity, new DeleteEntityIntention());
+        }
+
+        public void FinalizeComponents(in Query query)
+        {
+            World.InlineQuery<ReleaseOnEntityDestroy, SDKAvatarShapeComponent>(new QueryDescription().WithAll<SDKAvatarShapeComponent>(), ref releaseOnEntityDestroy);
```

Comment:
You are not obliged to use structure-like query handlers. In other places, it was reused across multiple queries. Here it looks like an unneeded boiler-plate, consider using a code-generated way.

Reply:
I updated it to `World.Query(new QueryDescription().WithAll<SDKAvatarShapeComponent>(), (ref SDKAvatarShapeComponent removedComponent) => globalWorld.Add(removedComponent.globalWorldEntity, new DeleteEntityIntention()));` that's better right?

Reply:
No, it's worse 😄 The delegate notation is the slowest

Reply:
Done!